internal/server/data/provideruser.go

```diff
@@ -236,8 +246,8 @@ func SyncProviderUser(ctx context.Context, tx WriteTxn, user *models.Identity, p
 }
 
 type SCIMParameters struct {
-	Count      int // the number of items to return
-	StartIndex int // the offset to start counting from
-	TotalCount int // the total number of items that match the query
-	// TODO: filter query param
+	Count      int                // the number of items to return
+	StartIndex int                // the offset to start counting from
+	TotalCount int                // the total number of items that match the query
+	Filter     *filter.Expression // a filter to apply to the results
```

Comment:
`filter.Expression` is an interface, so I think this should not be a pointer. An interface allows us to do `!= nil` without it being a pointer.

```suggestion
	Filter     filter.Expression // a filter to apply to the results
```